src/git-repository-async.js

```diff
@@ -0,0 +1,871 @@
+  //
+  // Note that if the status of the path has changed, this will emit a
+  // 'did-change-status' event.
+  //
+  // * `path` The {String} path whose status should be refreshed.
+  //
+  // Returns a {Promise} which resolves to a {Number} which is the refreshed
+  // status bit for the path.
+  refreshStatusForPath (_path) {
+    this._refreshingCount++
+
+    let relativePath
+    return this.repoPromise
+      .then(repo => {
+        relativePath = this.relativize(_path, repo.workdir())
+        return this._filterStatusesByPath(_path)
+      })
+      .then(statuses => {
+        const cachedStatus = this.pathStatusCache[relativePath] || 0
+        const status = statuses[0] ? statuses[0].statusBit() : Git.Status.STATUS.CURRENT
+        if (status !== cachedStatus) {
+          this.pathStatusCache[relativePath] = status
+          this.emitter.emit('did-change-status', {path: _path, pathStatus: status})
+        }
+
+        return status
+      })
+      .then(status => {
+        this._refreshingCount--
+        return status
+      })
+  }
+
+  // Returns a Promise that resolves to the status bit of a given path if it has
+  // one, otherwise 'current'.
+  getPathStatus (_path) {
+    return this.refreshStatusForPath(_path)
+  }
+
+  // Public: Get the cached status for the given path.
+  //
+  // * `path` A {String} path in the repository, relative or absolute.
+  //
+  // Returns a {Promise} which resolves to a status {Number} or null if the
+  // path is not in the cache.
+  getCachedPathStatus (_path) {
+    return this.repoPromise
+      .then(repo => this.relativize(_path, repo.workdir()))
+      .then(relativePath => this.pathStatusCache[relativePath])
+  }
+
+  // Public: Returns true if the given status indicates modification.
+  //
+  // * `statusBit` A {Number} representing the status.
+  //
+  // Returns a {Boolean} that's true if the `statusBit` indicates modification.
+  isStatusModified (statusBit) {
+    return (statusBit & modifiedStatusFlags) > 0
+  }
+
+  // Public: Returns true if the given status indicates a new path.
+  //
+  // * `statusBit` A {Number} representing the status.
+  //
+  // Returns a {Boolean} that's true if the `statusBit` indicates a new path.
+  isStatusNew (statusBit) {
+    return (statusBit & newStatusFlags) > 0
+  }
+
+  // Public: Returns true if the given status indicates the path is staged.
+  //
+  // * `statusBit` A {Number} representing the status.
+  //
+  // Returns a {Boolean} that's true if the `statusBit` indicates the path is
+  // staged.
+  isStatusStaged (statusBit) {
+    return (statusBit & indexStatusFlags) > 0
+  }
+
+  // Public: Returns true if the given status indicates the path is ignored.
+  //
+  // * `statusBit` A {Number} representing the status.
+  //
+  // Returns a {Boolean} that's true if the `statusBit` indicates the path is
+  // ignored.
+  isStatusIgnored (statusBit) {
+    return (statusBit & (1 << 14)) > 0
```

Comment:
Super minor, worth pulling `1 << 14` to a `ignoredStatusFlags` constant for consistency?